tests/sqlparser_postgres.rs

```diff
@@ -2812,6 +2812,19 @@ fn test_json() {
     );
 }
 
+#[test]
+fn test_json_object() {
```

Comment:
```suggestion
fn test_fn_arg_with_value_operator() {
```
thinking or something more generic? since the feature can apply in other cases and json_object has other variants

Reply:
I applied your suggestion. But this feature really should not apply to other functions. The syntax `'key' VALUE 'value'` is specific to json_object and will not be parsed by actual databases in arbitrary functions. The decision to parse this as a generic function call instead of a particular expression was taken in https://github.com/apache/datafusion-sqlparser-rs/pull/1507#discussion_r1835440360 , so I did not come back on the design.